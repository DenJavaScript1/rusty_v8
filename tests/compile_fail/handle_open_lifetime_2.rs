// Copyright 2019-2020 the Deno authors. All rights reserved. MIT license.

pub fn main() {
  let isolate = &mut v8::Isolate::new(mock());
  let context: v8::Local<v8::Context> = mock();
  let scope = &mut v8::HandleScope::with_context(isolate, context);

  let local1 = v8::Object::new(scope);
  let global1 = v8::Global::new(scope, local1);
  let local2 = v8::Handle::open(&global1, scope);

  drop(global1);
  let _global2 = v8::Global::new(scope, local2);
}

fn mock<T>() -> T {
  unimplemented!()
}
