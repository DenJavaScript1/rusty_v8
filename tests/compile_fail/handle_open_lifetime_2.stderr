error[E0505]: cannot move out of `global1` because it is borrowed
  --> tests/compile_fail/handle_open_lifetime_2.rs:12:8
   |
10 |   let local2 = v8::Handle::open(&global1, scope);
   |                                 -------- borrow of `global1` occurs here
11 |
12 |   drop(global1);
   |        ^^^^^^^ move out of `global1` occurs here
13 |   let _global2 = v8::Global::new(scope, local2);
   |                                         ------ borrow later used here
